lib/rule-context.js

```diff
@@ -161,22 +162,14 @@ class RuleContext {
         this.parserServices = Object.freeze(Object.assign({}, parserServices));
 
         // private.
-        this.eslint = eslint;
-        this.severity = severity;
+        this._linter = linter;
```

Comment:
@mysticatea Will this break `eslint-plugin-eslint-comments`? I recall it was relying on `context.eslint` at one point.

(Ideally, I think it would be better if rules didn't have access to the `linter` instance, because it allows them to interfere with other rules. But I'm not sure there's a better way to implement `eslint-plugin-eslint-comments`, except possibly by running the rules with a separate `Linter` instance).